crates/swift-integration-tests/src/result.rs

```diff
@@ -36,6 +36,18 @@ mod ffi {
 
         fn val(&self) -> u32;
     }
+
+    enum ResultTransparentEnum {
+        NamedField { data: i32 },
+        UnnamedFields(u8, String),
+        NoFields,
+    }
+
+    extern "Rust" {
+        fn rust_func_return_result_transparent_enum_opaque_rust(
+            succeed: bool,
+        ) -> Result<ResultTestOpaqueRustType, ResultTransparentEnum>;
```

Comment:
Does `Result<Enum, Enum>` work?

If so, can we add a test for that?

Can we also test `Result<Enum, OpaqueRust>`, since that's a fairly common pattern in Rust code?

Reply:
Can we also add a `fn same_custom_result_returned_twice_first() -> Result<Enum, Enum>` `fn same_custom_result_returned_twice_second() -> Result<Enum, Enum>` 

We don't need to call either of these from Swift. They're just here to confirm that we don't generate the same types twice.